EmailLogCleanup/src/main/java/org/vm/shared/S3LogWriter.java

```diff
@@ -0,0 +1,50 @@
+package org.vm.shared;
+
+import com.amazonaws.SdkClientException;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+
+import java.io.InputStream;
+
+/**
+ * The S3LogWriter class takes is a class that implements the WriteLog interface and specifiying it
+ * to write to a file to an s3 bucket.
+ */
+public class S3LogWriter implements WriteLog{
+  
+  private final String bucket;
+  
+  /**
+   * The constructor takes in the name of the destination S3 bucket.
+   * @param bucketName is the name of the destination bucket in AWS.
+   */
+  public S3LogWriter(String bucketName) {
+    bucket = bucketName;
+  }
+  
+  /**
+   * Method inherited from the WriteLog interface that will create a put request for a file
+   * into an s3 bucket.
+   */
+  @Override
+  public void writeLog(InputStream stream, String filename) {
+    //Sets up the s3 bucket.
+    AmazonS3 s3Client = AmazonS3ClientBuilder.defaultClient();
+  
+    //Creates the PutObject request.
+    try {
+      PutObjectRequest request
+          = new PutObjectRequest(bucket, filename, stream, new ObjectMetadata());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: When objects are uploaded to Amazon S3 by using streams (either through an AmazonS3 client or TransferManager), you might encounter network connectivity or timeout issues.

**Fix**: The most reliable way to avoid a ResetException is to provide data by using a File or FileInputStream, which the AWS SDK for Java can handle without being constrained by mark and reset limits. If the input stream isn&#39;t a FileInputStream but does support mark and reset, you can set the mark limit by using the [setReadLimit](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/RequestClientOptions.html#setReadLimit-int-) method of [RequestClientOptions.](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/RequestClientOptions.html) Its default value is 128 KB. Setting the read limit value to one byte greater than the size of stream will reliably avoid a ResetException.

[Learn more](https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/best-practices.html)